src/components/SelectionList.tsx

```diff
@@ -108,14 +108,20 @@ const SelectionList = (props: ListProperties): React.JSX.Element => {
               <TouchableOpacity
                 onPress={() => {
                   if (props.type === 'single') {
+                    (props.setSelected as (d: Data) => void)(item);
                     (props.onSelect as (d: Data) => void)(item);
                     props.hide();
                   } else {
-                    const selected = props.selected as Data[],
-                      newSelected = selected.includes(item)
-                        ? selected.filter((d: Data) => d !== item)
-                        : [...selected, item];
-                    (props.onSelect as (d: Data[]) => void)(newSelected);
+                    const selected = props.selected as Data[];
+                    let newSelected: Data[];
+                    if (selected.includes(item)) {
+                      newSelected = selected.filter((d: Data) => d !== item);
+                      props.onRemove?.(item);
+                    } else {
+                      newSelected = [...selected, item];
+                      (props.onSelect as (d: Data[]) => void)(newSelected);
+                    }
+                    (props.setSelected as (d: Data[]) => void)(newSelected);
                   }
                 }}
```

Comment:
Is the UI update order reversed for single and multi-select on purpose? I think we should prefer the same order of function for single and multi-select, for consistency.

In single-select:
```
props.setSelected // updates ui
props.onSelect // runs action
```

In multi-select:
```
props.onSelect or props.onRemove // runs action
props.setSelected // updates ui
```